crates/bevy_audio/src/audio_output.rs

```diff
@@ -74,8 +89,99 @@ where
+    let mut sinks = world.get_resource_mut::<Assets<AudioSink>>().unwrap();
 
     if let Some(audio_sources) = world.get_resource::<Assets<Source>>() {
-        audio_output.try_play_queued(&*audio_sources, &mut *audio);
+        audio_output.try_play_queued(&*audio_sources, &mut *audio, &mut *sinks);
     };
 }
+
+/// Asset controlling the playback of a sound
+///
+/// ```
+/// # use bevy_ecs::system::{Local, Res};
+/// # use bevy_asset::{Assets, Handle};
+/// # use bevy_audio::AudioSink;
+/// fn pause(
+///     audio_sinks: Res<Assets<AudioSink>>,
+///     music_controler: Local<Handle<AudioSink>>,
+/// ) {
+///     if let Some(sink) = audio_sinks.get(&*music_controler) {
+///         if sink.is_paused() {
+///             sink.play()
+///         } else {
+///             sink.pause()
+///         }
+///     }
+/// }
+/// ```
+///
+#[derive(TypeUuid)]
+#[uuid = "8BEE570C-57C2-4FC0-8CFB-983A22F7D981"]
+pub struct AudioSink {
+    // This is an option to have a safe drop that will detach the sink.
+    // It will never be None during its life
+    sink: Option<Sink>,
+}
+
+impl Drop for AudioSink {
+    fn drop(&mut self) {
+        self.sink.take().unwrap().detach();
+    }
+}
+
+impl AudioSink {
+    /// Gets the volume of the sound.
+    ///
+    /// The value `1.0` is the "normal" volume (unfiltered input). Any value other than `1.0`
+    /// will multiply each sample by this value.
+    pub fn volume(&self) -> f32 {
+        self.sink.as_ref().unwrap().volume()
+    }
+
+    /// Changes the volume of the sound.
+    ///
+    /// The value `1.0` is the "normal" volume (unfiltered input). Any value other than `1.0`
+    /// will multiply each sample by this value.
+    pub fn set_volume(&self, volume: f32) {
+        self.sink.as_ref().unwrap().set_volume(volume);
+    }
+
+    /// Gets the speed of the sound.
+    ///
+    /// The value `1.0` is the "normal" speed (unfiltered input). Any value other than `1.0`
+    /// will change the play speed of the sound.
+    pub fn speed(&self) -> f32 {
+        self.sink.as_ref().unwrap().speed()
+    }
+
+    /// Changes the speed of the sound.
+    ///
+    /// The value `1.0` is the "normal" speed (unfiltered input). Any value other than `1.0`
+    /// will change the play speed of the sound.
+    pub fn set_speed(&self, speed: f32) {
+        self.sink.as_ref().unwrap().set_speed(speed);
+    }
+
+    /// Resumes playback of a paused sink.
+    ///
+    /// No effect if not paused.
+    pub fn play(&self) {
+        self.sink.as_ref().unwrap().play();
+    }
+
+    /// Pauses playback of this sink.
+    ///
+    /// No effect if already paused.
+    /// A paused sink can be resumed with [`play`](Self::play).
+    pub fn pause(&self) {
+        self.sink.as_ref().unwrap().pause();
+    }
+
+    /// Gets if a sink is paused
+    ///
+    /// Sinks can be paused and resumed using [`pause`](Self::pause) and [`play`](Self::play). This returns true if the sink is paused.
```

Comment:
```suggestion
    /// Sinks can be paused and resumed using [`pause`](Self::pause) and [`play`](Self::play).
```

Reply:
same https://docs.rs/rodio/latest/rodio/struct.Sink.html#method.is_paused 😄 

I'm open to change it, but I like having the same api and doc as the underlying rodio

Reply:
IMO we shouldn't couple tightly to rodio; it has its problems and it will make maintaining this code much more frustrating without good reason.